x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/role/RestGetRolesAction.java

```diff
@@ -57,25 +54,43 @@ public String getName() {
     @Override
     public RestChannelConsumer innerPrepareRequest(RestRequest request, NodeClient client) throws IOException {
         final String[] roles = request.paramAsStringArray("name", Strings.EMPTY_ARRAY);
-        return channel -> new GetRolesRequestBuilder(client).names(roles).execute(new RestBuilderListener<>(channel) {
-            @Override
-            public RestResponse buildResponse(GetRolesResponse response, XContentBuilder builder) throws Exception {
-                builder.startObject();
-                for (RoleDescriptor role : response.roles()) {
-                    builder.field(role.getName(), role);
-                }
-                builder.endObject();
+        final boolean restrictRequest = request.hasParam(RestRequest.RESPONSE_RESTRICTED);
+        assert false == restrictRequest || DiscoveryNode.isStateless(settings);
+        return channel -> new GetRolesRequestBuilder(client).names(roles)
+            .nativeOnly(restrictRequest)
```

Comment:
We could consider exposing a `native_only` (or similar) request parameter publicly. In stateful, users could freely set this. In stateless, we'd restrict the ability to set it to operator users. Don't think we need to worry about this in the initial iteration though.

Reply:
Agreed on usefulness. Let's expose it if there is a need for it.